pkg/cloud/services/networking/port.go

```diff
@@ -91,9 +91,22 @@ func (s *Service) GetOrCreatePort(eventObject runtime.Object, clusterName string
 		}
 
 		securityGroups = portOpts.SecurityGroups
+		securityGroupFilters := portOpts.SecurityGroupFilters
 
+		if securityGroupFilters != nil {
+			securityGroupFiltersByID, err := s.GetSecurityGroups(securityGroupFilters)
+			if err != nil {
+				return nil, fmt.Errorf("error getting security groups: %v", err)
+			}
+			for _, sg := range securityGroupFiltersByID {
+				if sg == "" {
+					continue
+				}
+				*securityGroups = append(*securityGroups, sg)
```

Comment:
How will this handle duplicate security groups? E.g. if the same uuid is specified in `portOpts.SecurityGroups` and `portOpts.SecurityGroupFilters`

Reply:
Tried duplicates, and it does fail, no check on that.
Thanks, will add the logic to remove the duplicates.

Reply:
Done: now duplicates are removed
1. securityGroups field
2. securityGroupFilter field
3. A combination of the two.